src/QsCompiler/Compiler/ExternalRewriteSteps.cs

```diff
@@ -17,50 +17,88 @@ namespace Microsoft.Quantum.QsCompiler
 {
     internal static class RewriteSteps
     {
-        /// <summary>
-        /// Concrete implementation of the rewrite steps options such that the configured options may be null. 
-        /// </summary>
-        internal class RewriteStepOptions : IRewriteStepOptions
-        {
-            public readonly string OutputFolder;
-            internal RewriteStepOptions(CompilationLoader.Configuration config) =>
-                this.OutputFolder = config.BuildOutputFolder;
-        }
-
         /// <summary>
         /// Concrete implementation of a rewrite steps with an additional property specifying the dll it was loaded from. 
         /// </summary>
         internal class LoadedStep : IRewriteStep
```

Comment:
Have you considered using class template to get the IRewriteStep methods and fields?